Kconfig.zephyr

```diff
@@ -22,10 +22,10 @@ endmenu
 # $ARCH and $BOARD_DIR will be glob patterns when building documentation.
 source "boards/shields/*/Kconfig.defconfig"
 source "$(BOARD_DIR)/Kconfig.defconfig"
-source "$(SOC_DIR)/$(ARCH)/*/Kconfig.defconfig"
+source "$(KCONFIG_BINARY_DIR)/Kconfig.soc.defconfig"
 
 source "boards/Kconfig"
-source "$(SOC_DIR)/Kconfig"
```

Comment:
I see SOC_DIR still mentioned in doc/application/index.rst; is that OK?

Are you planning to update documentation later if this approach is accepted? If so, maybe consider adding DNM or marking as a draft.

I still see SOC_DIR getting set in boilerplate.cmake, so I'm not sure why it's not available for use here. I see that it's no longer set in the environment in kconfig.cmake, but I don't understand why if we still know where it is.

Reply:
yes, the docs still need to be updated, and thanks for the heads-up on SOC_DIR in the docs.
Will keep an extra eye on that.

I'm simply waiting for first feedback before updating the docs, especially on the yaml changes, as those are most important in that regard.

> I still see SOC_DIR getting set in boilerplate.cmake, so I'm not sure why it's not available for use here. I see that it's no longer set in the environment in kconfig.cmake, but I don't understand why if we still know where it is.

Because Kconfig determines the soc we are using, and once we know the soc, then we can set the soc dir.
But that means we need all soc roots available for Kconfig.

But for safety, the `${SOC_DIR}` should only be needed for `add_subdirectory` in CMake.
(it's used a limited number of places for includes and linker script, but hopefully that can be cleaned up later)